code/clue.js

```diff
@@ -1,59 +1,304 @@
+
+  document.getElementById("loader2").style.display = "none"
+  document.getElementById('weaponCard').removeEventListener('click', pickWeaponLoader);
+
+  document.getElementById("weaponCard").style.background = mystery.weapon.color;
+  document.getElementById("weaponName").innerHTML = `${mystery.weapon.name}`;
+  document.getElementById("weaponImage").src = mystery.weapon.image;
+  document.getElementById("weaponLastSeen").innerHTML = `Last seen in the ${mystery.weapon.lastseen}`;
+  document.getElementById("weaponStrength").innerHTML = `Strength: ${mystery.weapon.strength}`;
+
+  document.getElementById("cardDetails2").style.display = "flex";
+};
+
+//----- Room card -----//
+const pickRoomLoader = () => {
+  document.getElementById("loader3").classList.add("animated-loader");
+  setTimeout(showPickRoom, 2000);
+};
+
+document.getElementById("roomCard").addEventListener("click", pickRoomLoader); 
+
+const showPickRoom = () => {
+  mystery.room = randomSelector(rooms);
+  document.getElementById("loader3").style.display = "none"
+  document.getElementById('roomCard').removeEventListener('click', pickRoomLoader);
+
+  document.getElementById("roomCard").style.background = mystery.room.color;
+  document.getElementById("roomName").innerHTML = `${mystery.room.name}`;
+  document.getElementById("roomImage").src = mystery.room.image;
+  document.getElementById("roomLastSeen").innerHTML = `Last seen there: ${mystery.room.lastseen}`;
+  document.getElementById("roomfave").innerHTML = `Favourite room: ${mystery.room.favespot}`;
+
+  document.getElementById("cardDetails3").style.display = "flex";
+};
+
+//----- Function for on button click reveal mystery or get a message asking you to choose each card to then beable to reveal the mystery-----//
+function revealMystery() {
+  if (mystery.killer.hasOwnProperty("name") === true) {
+    document.getElementById("mystery").innerHTML = `The murder was commited by ${mystery.killer.name} with the ${mystery.weapon.name.toLowerCase()} in the ${mystery.room.name.toLowerCase()}`;
+    document.getElementById("revealMysteryButton").innerHTML = "PLAY AGAIN!";
+    document.getElementById("revealMysteryButton").onclick = resetPage;
+    document.getElementById("choose-card-prompt").style.display = "none";
+
+  } else {
+    document.getElementById("choose-card-prompt").innerHTML = "Please select one of each card to reveal the mystery!";
+  }
+};
 
-// This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+//----- Shuffle function -----//
+// Allows for the weapons to be shuffled between the suspects
+const shuffleFavouriteWeapon = () => {
+  let copyWeapons = weapons.slice(0); //Creates shallow copy of weapons array
+  for (let i = 0; i < suspects.length; i++ ) {
```

Comment:
very nice Claire :)

Reply:
Thank you @EmOlsz but I cannot take all the credit. Got some help from my partner on this one!